### ISSUE TITLE
Consolidate package-level & receiver-type suite Run() function

### ISSUE THREAD

**Issue by boyan-soubachov** [stretchr/testify#978]
# What?
The `suite.Run()` method exists both on the package level as well as a receiver of a suite type. The package-level function has a lot more development and features whereas the receiver is very basic and neglected.

# Why?
Reduce code duplication, remove user confusion over which one to use, reduce errors, centralise code.

# Risk
This is potentially a breaking change and could be incompatible with all existing code that uses the receiver-type function should we decide to remove it.

**Comment by mvdkleijn** [comment]
This would require proper documentation in the release notes at the very least. It'd also be nice to get a sense of how many people use the package level `suite.Run()` vs the receiver level one... though I doubt we'll be able to get that info.

**Comment by boyan-soubachov** [comment]
> This would require proper documentation in the release notes at the very least. It'd also be nice to get a sense of how many people use the package level `suite.Run()` vs the receiver level one... though I doubt we'll be able to get that info.

Agreed, very good points. I've been thinking about the whole testify v2 project and how we could efficiently extract some data on its usage from GitHub repos. Going to try and hack on it a bit in my (very limited 😞) spare time; but I think it would be very useful to get some hard numbers.
